agimus_controller/croco_hpp.py

```diff
@@ -3,12 +3,43 @@
 from .ocp_analyzer import *
 
 
+class TrajectoryBuffer:
+    def __init__(self, model):
+        self.model = model
+        self.x_plan = []
+        self.a_plan = []
+
+    def add_trajectory_point(self, q, v, a=None):
+        if len(q) != self.model.nq:
+            raise Exception(
+                f"configuration vector size : {len(q)} doesn't match model's nq : {self.model.nq}"
+            )
+        if len(v) != self.model.nv:
+            raise Exception(
+                f"velocity vector size : {len(v)} doesn't match model's nv : {self.model.nv}"
+            )
+        x = np.concatenate([np.array(q), np.array(v)])
+        self.x_plan.append(x)
+        if a is not None:
+            if len(a) != self.model.nv:
+                raise Exception(
+                    f"acceleration vector size : {len(a)} doesn't match model's nv : {self.model.nv}"
+                )
+            self.a_plan.append(np.array(a))
+
+    def get_x_plan(self):
```

Comment:
the name of the function is not extremely clear here.
X can be: [q, v] but also [q, v, a] or even [q, v, tau]...
I think a more explicit getter name is preferable.
I am open to suggestion.
I would go for `get_joint_position_horizon()`?
Sam for all quantities you get from HPP.